astroid/builder.py

```diff
@@ -237,6 +237,8 @@ def delayed_assattr(self, node):
                 try:
                     if inferred.__class__ is bases.Instance:
                         inferred = inferred._proxied
```

Comment:
I would like to see a test first with this problem, maybe it will have another fix than ignoring the node.